File: server/src/domain/model/jar.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type Jar struct {
	ID         int       `json:"id"`
	UserID     uuid.UUID `json:"user_id"`
	Name       string    `json:"name"`
	Percentage float64   `json:"percentage"`
	Balance    float64   `json:"balance"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}

File: client/lib/features/jars/domain/entities/jar.dart
import 'package:equatable/equatable.dart';

class Jar extends Equatable {
  final int? id;
  final String userId;
  final String name;
  final double percentage;
  final double balance;
  final DateTime createdAt;
  final DateTime? updatedAt;

  const Jar({
    this.id,
    required this.userId,
    required this.name,
    required this.percentage,
    required this.balance,
    required this.createdAt,
    this.updatedAt,
  });

  @override
  List<Object?> get props => [
    id,
    userId,
    name,
    percentage,
    balance,
    createdAt,
    updatedAt,
  ];
}
